### ISSUE TITLE
Different validation loss behavior between NVIDIA A40 and A100 GPUs with identical data and configuration

### ISSUE THREAD

**Issue by lisherlock** [MIC-DKFZ/nnUNet#2979]
Thanks for your great work!

I am training the same nnUNetv2 model on two different GPU types (NVIDIA A40 and NVIDIA A100-SXM4-80GB) with identical datasets, splits, and configurations. However, I observe significantly different training behaviors, particularly in the validation loss.

Environment

| | A40 | A100 |
|---|---|---|
| **GPU** | NVIDIA A40 | NVIDIA A100-SXM4-80GB |
| **PyTorch** | 2.0.1+cu117 | 2.0.1+cu117 |
| **cuDNN** | 8500 | 8500 |
| **nnUNet Trainer** | nnUNetTrainer_2000epochs_NoMirroring | nnUNetTrainer_2000epochs_NoMirroring |
| **Configuration** | 3d_fullres | 3d_fullres |

Training Configuration

- Batch size: 2
- Patch size: [112, 128, 112]
- Initial LR: 0.01
- Optimizer: SGD (momentum=0.99, nesterov=True, weight_decay=3e-05)
- Loss: DC_and_CE_loss with DeepSupervisionWrapper

Observed Behavior

**On A40 (works correctly):**
- Training loss decreases and converges to ~0
- Validation loss decreases and converges to ~0
- Pseudo Dice reaches ~0.85
- No sign of overfitting

**On A100 (problematic):**
- Training loss decreases and converges to ~0 (same as A40)
- **Validation loss stays high at 0.15-0.2 and does not converge**
- Pseudo Dice reaches ~0.80
- Shows apparent overfitting pattern (large gap between train/val loss)

My hypothesis

I suspect this might be related to:
1. **TensorFloat-32 (TF32)** precision differences on A100
2. **Mixed precision training (AMP)** behaving differently on A100's Tensor Cores
3. **cuDNN algorithm selection** differences between GPU architectures

Questions

1. Is this a known issue when training nnUNetv2 on A100 GPUs?
2. Are there any recommended settings or environment variables to ensure consistent behavior across different GPU types?
3. Should I disable TF32 or use deterministic cuDNN settings for reproducibility?

<img width="3000" height="5400" alt="Image" src="https://github.com/user-attachments/assets/bd1c0250-ba46-4273-9104-330a9a3c3969" />

<img width="1455" height="927" alt="Image" src="https://github.com/user-attachments/assets/1364426f-ebe8-493a-b6f9-5c60aaab6592" />

Thank you for your help!
